src/platform/packages/shared/kbn-alerts-ui-shared/src/alert_rule_from_vis_ui_action/index.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { pick } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import type { Action } from '@kbn/ui-actions-plugin/public';
+import { ALERT_RULE_TRIGGER } from '@kbn/ui-actions-browser/src/triggers';
+import { apiIsOfType, hasBlockingError } from '@kbn/presentation-publishing';
+import type { LensApi } from '@kbn/lens-plugin/public';
+import type { TextBasedPersistedState } from '@kbn/lens-plugin/public/datasources/text_based/types';
+import type { Datatable } from '@kbn/expressions-plugin/common';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { ActionTypeModel, RuleTypeModel } from '../common';
+import type { TypeRegistry } from '../common/type_registry';
+
+export interface AlertRuleFromVisUIActionData {
+  timeField: string;
+  query: string | null;
+  thresholdValues: Record<string, number>;
+  splitValues: Record<string, Array<string | number | null | undefined>>;
+  usesPlaceholderValues?: boolean;
+  dataView?: DataView;
+}
+
+interface Context {
+  data?: AlertRuleFromVisUIActionData;
+  embeddable: LensApi;
+}
+
+export class AlertRuleFromVisAction implements Action<Context> {
```

Comment:
Not sure if `alerts-ui-shared` is the right package for this, let me know if there's a better place to put it.